### DISCUSSION TITLE
Questions about auto-mode-alist settings

### DISCUSSION THREAD

**Discussion by eki3z** [magit/git-modes#142] [Support] [answered]
I have settings as below:
```elisp
(setq-default major-mode
              (lambda () (if buffer-file-name
                             (fundamental-mode)
                           (let ((buffer-file-name (buffer-name)))
                             (set-auto-mode)))))

```
when it's a file-visiting buffer, set default to `fundamental-mode`, when it's not, set major-mode according to `auto-mode-alist`. And it's doing well until I edit a new buffer ".gitconfig", it failed. I see these code

```elisp
;;;###autoload
(dolist (pattern '("/\\.gitconfig\\'"      "/\\.git/config\\'"
                   "/modules/.*/config\\'" "/git/config\\'"
                   "/\\.gitmodules\\'"     "/etc/gitconfig\\'"))
  (add-to-list 'auto-mode-alist (cons pattern 'gitconfig-mode)))
```

Of course it's no problem ,  but I just doubt what difference between `"/\\.gitconfig\\'"` and `"\\.gitconfig\\'"` ?
with latter, my major-mdoe setting is OK, the former couldn't apply to a buffer which is no-file-visiting. 
Is there a improvement ?

**Comment by tarsius** [comment, marked as answer]
> I have settings as below:

I am not sure you are supposed to use the option like this but at least I didn't see anything that said otherwise.

> Of course it's no problem ,  but I just doubt what difference between `"/\\.gitconfig\\'"` and `"\\.gitconfig\\'"` ?

The first only matches a file named `.gitconfig`, the second also matches `whatever.gitconfig`.

**Reply by eki3z** [reply]
OK, thanks!

I just curious about which procedure you're doing when you edit a new buffer without automatic setting major-mode ? Call it manually or save it first ?

**Reply by tarsius** [reply]
I don't quite understand what you are asking but get a feeling this is a general Emacs question and recommend you use e.g. the Emacs Reddit for such questions.

**Reply by eki3z** [reply]
Thanks !
